lib/trip.rb

```diff
@@ -0,0 +1,67 @@
+require 'csv'
+
+module RideShare
+  class Trip
+    attr_accessor :trip_id, :rider_id, :driver_id, :date, :rating
+
+    def initialize(trip_id, driver_id, rider_id, date, rating)
+      raise ArgumentError.new ("rating must be between 1-5") if rating > 5 || rating < 1
+      @trip_id = trip_id
+      @driver_id = driver_id
+      @rider_id = rider_id
+      @date = date
+      @rating = rating
+    end
+
+    # retrieve the associated driver instance through the driver ID
+    def driver_object
+      return RideShare::Driver.find_driver(@driver_id)
+    end
+
+    # retrieve the associated rider instance through the rider ID
+    def rider_object
+      return RideShare::Rider.find_rider(@rider_id)
+    end
+
+    # retrieve all trips from the CSV file
+    def self.all_trip_info
+      all_trips_array = []
+      CSV.read('support/trips.csv').each do |object|
+        trip_id = object[0].to_i
+        driver_id = object[1].to_i
+        rider_id = object[2].to_i
+        date = object[3].to_s
+        rating = object[4].to_f
+        a_trip = RideShare::Trip.new(trip_id, driver_id, rider_id, date, rating)
+        all_trips_array << a_trip
+      end
+      return all_trips_array
+      # return all the trips in the trip csv
+    end
+
+    #all trip instances with a particular driver id
+    def self.driver_trip_instances(driver_id)
+      driver_trips = []
+      trips = RideShare::Trip.all_trip_info
+      trips.each do |object|
```

Comment:
Could you clean this up a little using an enumerable? Maybe `select` would be a good option.